src/base/random.hh

```diff
@@ -102,18 +101,20 @@ class Random : public Serializable
     random()
     {
         // [0, 1) for real types
-        std::uniform_real_distribution<T> dist;
-        return dist(gen);
+        return (T) random<uint64_t>() /
+          (T) std::numeric_limits<uint64_t>::max();
     }
+
     /**
      * @ingroup api_base_utils
      */
     template <typename T>
     typename std::enable_if_t<std::is_integral_v<T>, T>
     random(T min, T max)
     {
-        std::uniform_int_distribution<T> dist(min, max);
-        return dist(gen);
+        assert(min < max);
+        T r = (gen() - min) % (max - min) + min;
```

Comment:
Maybe it already exists, but it would be good to test this with small numbers to make sure the inclusive/exclusive on the limits works. I.e., `random(0, 2)` should return 0, 1, and 2.

Reply:
Done (see random.test.cc, last test)

Reply:
For copyright, you'll have to ask your institution. For instance, here is my institution's policy: https://copyright.universityofcalifornia.edu/resources/copyright-ownership-faqs.html

Reply:
It's the same as Inria which has some copyright mentions in headers e.g. src/cpu/pred/tage_base.hh, so I think it's as it should be.